Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs

```diff
@@ -133,3 +133,5 @@ private void SetSizeAfterUpdate(int longestEntryLength, int contentCount)
         }
     }
 }
+
+public record VendorItemsListData(EntityUid Dummy, string ItemName, int ItemIndex) : ListData;
```

Comment:
is there a reason why this still needs to spawn dummy entities on the client? EntityPrototypeView handles this independently so i don't know why you're storing a reference here.

Reply:
Made it use ``EntProtoId`` instead of spawning dummy entities.